d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/TrackerClientState.java

```diff
@@ -0,0 +1,116 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.d2.balancer.strategies.relative;
+
+import com.linkedin.util.degrader.ErrorType;
+import java.util.Map;
+
+
+/**
+ * Keeps the state of each tracker client for a partition
+ */
+public class TrackerClientState {
+  private int _callCount;
+  // TODO: update adjusted min call count
+  private int _adjustedMinCallCount;
+  private double _healthScore;
+  private boolean _isUnhealthy;
+  private boolean _isHealthy;
```

Comment:
redundant. also does this class need to know if it's healthy or not? Shouldn't the state updater calculate that and use it? Why temporarily store in this class?

Reply:
I changed to use an enum as you suggested during out meeting